torch/nn/parallel/distributed.py

```diff
@@ -366,15 +366,22 @@ class DistributedDataParallel(Module):
                                wrapped module's ``forward`` function. Parameters
                                that don't receive gradients as part of this
                                graph are preemptively marked as being ready to
-                               be reduced. Note that all ``forward`` outputs
-                               that are derived from module parameters must
-                               participate in calculating loss and later the
-                               gradient computation. If they don't, this wrapper
-                               will hang waiting for autograd to produce
-                               gradients for those parameters. Any outputs
-                               derived from module parameters that are otherwise
-                               unused can be detached from the autograd graph
-                               using ``torch.Tensor.detach``. (default: ``False``)
+                               be reduced. In addition, when running with
```

Comment:
My feeling is that the docs usually refer to the latest version of PyTorch and are coupled to the release itself. If so, we should probably just remove this from the docs which implicitly indicates this works now.

Maybe we can elaborate about this in our release notes mentioning this now works for DDP.